zio-http-cli/src/main/scala/zio/http/endpoint/cli/HttpCliApp.scala

```diff
@@ -0,0 +1,771 @@
+                  CliEndpoint[java.time.LocalDateTime](
+                    (
+                      localDateTime,
+                      request,
+                    ) => request.addFieldToBody(prefix, Json.Str(localDateTime.toString())),
+                    Options.localDateTime(prefix.mkString(".")), // FIXME
+                    List.empty,
+                    Doc.empty,
+                  ),
+                )
+              case StandardType.MonthType          =>
+                Set(
+                  CliEndpoint[String](
+                    (
+                      month,
+                      request,
+                    ) => request.addFieldToBody(prefix, Json.Str(month)),
+                    Options.text(prefix.mkString(".")), // FIXME
+                    List.empty,
+                    Doc.empty,
+                  ),
+                )
+              case StandardType.YearType           =>
+                Set(
+                  CliEndpoint[BigInt](
+                    (
+                      year,
+                      request,
+                    ) => request.addFieldToBody(prefix, Json.Num(BigDecimal(year))),
+                    Options.integer(prefix.mkString(".")), // FIXME
+                    List.empty,
+                    Doc.empty,
+                  ),
+                )
+            }
+          case Schema.Fail(_, _)                    => Set.empty
+          case Schema.Map(_, _, _)                  => ??? // TODO
+          case Schema.Sequence(_, _, _, _, _)       => ??? // TODO
+          case Schema.Set(_, _)                     => ??? // TODO
+          case Schema.Lazy(schema0)                 => loop(prefix, schema0())
+          case Schema.Dynamic(_)                    => Set.empty
+          case Schema.Either(left, right, _)        => loop(prefix, left) ++ loop(prefix, right)
+          case Schema.Optional(schema, _)           => loop(prefix, schema).map(_.optional)
+          case Schema.Tuple2(left, right, _)        =>
+            for {
+              l <- loop(prefix, left)
+              r <- loop(prefix, right)
+            } yield l ++ r
+          case Schema.Transform(schema, _, _, _, _) => loop(prefix, schema)
+        }
+
+      loop(List.empty, schema)
+    }
+  }
+
+  /**
+   * Generates a [[zio.cli.CliApp]] from the given endpoints.
+   *
+   * @param name
+   *   The name of the generated CLI
+   * @param version
+   *   The version of the generated CLI
+   * @param summary
+   *   The summary of the generated CLI
+   * @param footer
+   *   Footer for the help docs of the generated CLI
+   * @param endpoints
+   *   Chunk of [[Endpoint]]
+   * @param host
+   *   The server host where the generated CLI will send requests to
+   * @param port
+   *   The server port where the generated CLI will send requests to
+   * @return
+   *   a [[zio.cli.CliApp]]
+   */
+  def fromEndpoints[M <: EndpointMiddleware](
+    name: String,
+    version: String,
+    summary: String,
```

Comment:
If in `zio-cli`, any of these take more rich docs, then we could accept the richer types here, rather than just `String`, so you could do formatting if you wanted to. The user can always lift a string to the richer doc.